website/static/js/share/utils.js

```diff
@@ -190,30 +228,269 @@ utils.removeFilter = function(vm, filter){
     utils.search(vm);
 };
 
-utils.arrayEqual = function(a, b) {
+/* Tests array equality */
+utils.arrayEqual = function (a, b) {
     return $(a).not(b).length === 0 && $(b).not(a).length === 0;
 };
 
-utils.loadStats = function(vm){
-    vm.statsLoaded(false);
+utils.addFiltersToQuery = function (query, filters) {
+    if (filters) {
+        filters.forEach(function (filter) {
+            query = utils.filteredQuery(query, filter.filter);
+        });
+    }
+    return query;
+};
 
+/* Loads the raw and normalized data for a specific result */
+utils.loadRawNormalized = function(result){
+    var nonJsonErrors = function(xhr) {
```

Comment:
Right - it was because the missing API didn't return an error in JSON, rather an html 404. I got this from the mithril github thread on how to handle 404 errors that aren't in json by overriding extract  https://github.com/lhorie/mithril.js/issues/123